TACC/tup-ui
# fix: mfa pair validation error message

## Overview & Changes

1. Fix short inline errors to appear beneath input field
2. Improve user-friendliness of MFA validation error message.
3. Include technical error message via new pattern.

## Related

- [TUP-550](https://jira.tacc.utexas.edu/browse/TUP-550)

## Testing

0. Be unpaired.
1. Open http://localhost:8000/portal/mfa/sms or http://localhost:8000/portal/mfa/totp.
2. If at `/totp` (QR code), then generate QR code.
3. On either MFA method, verify gibberish token.
4. Confirm error message is
    > Your token is invalid.

## UI

| QR | SMS |
| - | - |
| ![qr](https://github.com/TACC/tup-ui/assets/62723358/720a2090-adee-47da-856e-aa0bddf80965) | ![sms](https://github.com/TACC/tup-ui/assets/62723358/94555516-3e0a-4a27-a5b5-404b0b280f86) |

<details><summary>ⓧ Include Technical Error (11aff00, reverted)</summary>

> **Important**
> Outdated solution. [Reason.](https://github.com/TACC/tup-ui/pull/296#pullrequestreview-1577587176)

| before clicking the error | after clicking the error |
| - | - |
| ![Error (before click)](https://github.com/TACC/tup-ui/assets/62723358/fa41d0c1-5537-485c-837d-191899b5475a) | ![Error (after click)](https://github.com/TACC/tup-ui/assets/62723358/3b9c23d4-2c3b-43a0-bca1-59a5a8dc7b0c) |

https://github.com/TACC/tup-ui/assets/62723358/06b3afea-00b0-4156-b8f3-63c139e10efb

</details>
